runtime/oti/arm64helpers.m4

```diff
@@ -82,41 +82,121 @@ define({CALL_C_WITH_VMTHREAD},{
 	CALL_DIRECT($1)
 })
 
-dnl No need to save/restore D8-D31 - the stack walker will never need to read
-dnl or modify them (no preserved FPRs in the JIT private linkage).
-
-dnl To be filled
 define({SAVE_C_VOLATILE_REGS},{
+	stp x0,x1,JIT_GPR_SAVE_SLOT(0)
+	stp x2,x3,JIT_GPR_SAVE_SLOT(2)
+	stp x4,x5,JIT_GPR_SAVE_SLOT(4)
+	stp x6,x7,JIT_GPR_SAVE_SLOT(6)
+	stp x8,x9,JIT_GPR_SAVE_SLOT(8)
+	stp x10,x11,JIT_GPR_SAVE_SLOT(10)
+	stp x12,x13,JIT_GPR_SAVE_SLOT(12)
+	stp x14,x15,JIT_GPR_SAVE_SLOT(14)
+	stp d0,d1,JIT_GPR_SAVE_SLOT(0)
+	stp d2,d3,JIT_GPR_SAVE_SLOT(2)
+	stp d4,d5,JIT_GPR_SAVE_SLOT(4)
+	stp d6,d7,JIT_GPR_SAVE_SLOT(6)
+	stp d16,d17,JIT_GPR_SAVE_SLOT(16)
+	stp d18,d19,JIT_GPR_SAVE_SLOT(18)
+	stp d20,d21,JIT_GPR_SAVE_SLOT(20)
+	stp d22,d23,JIT_GPR_SAVE_SLOT(22)
+	stp d24,d25,JIT_GPR_SAVE_SLOT(24)
+	stp d26,d27,JIT_GPR_SAVE_SLOT(26)
+	stp d28,d29,JIT_GPR_SAVE_SLOT(28)
+	stp d30,d31,JIT_GPR_SAVE_SLOT(30)
 })
 
-dnl To be filled
 define({RESTORE_C_VOLATILE_REGS},{
+	ldp x0,x1,JIT_GPR_SAVE_SLOT(0)
+	ldp x2,x3,JIT_GPR_SAVE_SLOT(2)
+	ldp x4,x5,JIT_GPR_SAVE_SLOT(4)
+	ldp x6,x7,JIT_GPR_SAVE_SLOT(6)
+	ldp x8,x9,JIT_GPR_SAVE_SLOT(8)
+	ldp x10,x11,JIT_GPR_SAVE_SLOT(10)
+	ldp x12,x13,JIT_GPR_SAVE_SLOT(12)
+	ldp x14,x15,JIT_GPR_SAVE_SLOT(14)
+	ldp d0,d1,JIT_GPR_SAVE_SLOT(0)
+	ldp d2,d3,JIT_GPR_SAVE_SLOT(2)
+	ldp d4,d5,JIT_GPR_SAVE_SLOT(4)
+	ldp d6,d7,JIT_GPR_SAVE_SLOT(6)
+	ldp d16,d17,JIT_GPR_SAVE_SLOT(16)
+	ldp d18,d19,JIT_GPR_SAVE_SLOT(18)
+	ldp d20,d21,JIT_GPR_SAVE_SLOT(20)
+	ldp d22,d23,JIT_GPR_SAVE_SLOT(22)
+	ldp d24,d25,JIT_GPR_SAVE_SLOT(24)
+	ldp d26,d27,JIT_GPR_SAVE_SLOT(26)
+	ldp d28,d29,JIT_GPR_SAVE_SLOT(28)
+	ldp d30,d31,JIT_GPR_SAVE_SLOT(30)
 })
 
-dnl To be filled
 define({SAVE_C_NONVOLATILE_REGS},{
+	stp x16,x17,JIT_GPR_SAVE_SLOT(16)
+	stp x18,x19,JIT_GPR_SAVE_SLOT(18)
+	stp x20,x21,JIT_GPR_SAVE_SLOT(20)
+	stp x22,x23,JIT_GPR_SAVE_SLOT(22)
+	stp x24,x25,JIT_GPR_SAVE_SLOT(24)
+	stp x26,x27,JIT_GPR_SAVE_SLOT(26)
+	stp x28,x29,JIT_GPR_SAVE_SLOT(28)
+	stp d8,d9,JIT_GPR_SAVE_SLOT(8)
+	stp d10,d11,JIT_GPR_SAVE_SLOT(10)
+	stp d12,d13,JIT_GPR_SAVE_SLOT(12)
+	stp d14,d15,JIT_GPR_SAVE_SLOT(14)
 })
 
-dnl To be filled
 define({RESTORE_C_NONVOLATILE_REGS},{
+	ldp x16,x17,JIT_GPR_SAVE_SLOT(16)
+	ldp x18,x19,JIT_GPR_SAVE_SLOT(18)
+	ldp x20,x21,JIT_GPR_SAVE_SLOT(20)
+	ldp x22,x23,JIT_GPR_SAVE_SLOT(22)
+	ldp x24,x25,JIT_GPR_SAVE_SLOT(24)
+	ldp x26,x27,JIT_GPR_SAVE_SLOT(26)
+	ldp x28,x29,JIT_GPR_SAVE_SLOT(28)
+	ldp d8,d9,JIT_GPR_SAVE_SLOT(8)
+	ldp d10,d11,JIT_GPR_SAVE_SLOT(10)
+	ldp d12,d13,JIT_GPR_SAVE_SLOT(12)
+	ldp d14,d15,JIT_GPR_SAVE_SLOT(14)
 })
 
 define({SAVE_ALL_REGS},{
 	SAVE_C_VOLATILE_REGS
 	SAVE_C_NONVOLATILE_REGS
 })
 
-dnl To be filled
-define({SAVE_PRESERVED_REGS},{
-})
-
 define({RESTORE_ALL_REGS},{
 	RESTORE_C_VOLATILE_REGS
 	RESTORE_C_NONVOLATILE_REGS
 })
 
-dnl To be filled
+define({SAVE_PRESERVED_REGS},{
```

Comment:
If my understanding is correct, `SAVE_PRESERVED_REGS` and `RESTORE_PRESERVED_REGS` refers to the JIT private linkage.  If you're not doing anything different from the ABI linkage then my comments above about special purpose registers apply here as well.

Reply:
Removed x16-x18 from preserved registers.
I will visit this again when I define the JIT private linkage.